added decontamination-authentication chapter (#1)
SPAAM-community/intro-to-ancient-metagenomics-book: decontamination-authentication.qmd

![](assets/images/chapters/decontamination-authentication/decontam.png)

Another popular tool for detecting contaminating microorganisms is [Recentrifuge](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1006967). It works as a classifier that is trained to recognize contaminant microbial organisms. In case of Recentrifuge, one has to use blanks or other negative controls and provide microbial names and abundances on the blanks in order to train Recentrifuge to recognize endogenous vs. contaminant sources.

> If one wants to assess the degree of contamination for each sample, there is a handy tool [cuperdec](https://github.com/jfy133/cuperdec), which is an R package that allows a quick comparison of microbial profiles in a query metagenomic sample against a database. The idea of *cuperdec* is to rank organisms in each sample by their abundance and then using an "expanding window" approach to compute their enrichment in a reference database that contains a comprehensive list of microbial organisms which are specific to a tissue / environment in question. The tool produces so-called *Cumulative Percent Decay* curves that aim to represent the level of endogenous content of microbiome samples, such as ancient dental calculus, to help to identify samples with low levels of preservation that should be discarded for downstream analysis.

Reviewer: 'expanding window' that's a great term1 I've always had such a hard time explaining these plots in words :face_exhaling:

> If one wants to assess the degree of contamination for each sample, there is a handy tool [cuperdec](https://github.com/jfy133/cuperdec), which is an R package that allows a quick comparison of microbial profiles in a query metagenomic sample against a database. The idea of *cuperdec* is to rank organisms in each sample by their abundance and then using an "expanding window" approach to compute their enrichment in a reference database that contains a comprehensive list of microbial organisms which are specific to a tissue / environment in question. The tool produces so-called *Cumulative Percent Decay* curves that aim to represent the level of endogenous content of microbiome samples, such as ancient dental calculus, to help to identify samples with low levels of preservation that should be discarded for downstream analysis.

Reviewer: ```suggestion
If one wants to assess the degree of contamination for each sample, there is a handy tool [cuperdec](https://github.com/jfy133/cuperdec), which is an R package that allows a quick comparison of microbial profiles in a query metagenomic sample against a database. The idea of `cuperdec` is to rank organisms in each sample by their abundance and then using an "expanding window" approach to compute their 'enrichment' of taxa from similar samples or environments, when comparing to a comprehensive list of microbial organisms which are specific to a tissue / environment in question. The tool produces so-called *Cumulative Percent Decay* curves that aim to represent the level of endogenous content of microbiome samples, such as ancient dental calculus, to help to identify samples with low levels of preservation that should be discarded for downstream analysis.
```

```R
library("cuperdec")
library("magrittr")
library("dplyr")

# Load database (in this case oral database)
data(cuperdec_database_ex)
database <- load_database(cuperdec_database_ex, target = "oral") %>% print()

# Load abundance matrix and metadata
taxatable <- load_taxa_table("abund_matrix.txt")  %>% print()
metadata <- as_tibble(data.frame(Sample = unique(taxatable$Sample), 
Sample_Source = "Oral"))

# Compute cumulative percent decay curves, filter and plot results
curves <- calculate_curve(taxatable, database = database) %>% print()
filter_result <- simple_filter(curves, percent_threshold = 50) %>% print()
plot_cuperdec(curves, metadata = my_metadata_map, filter_result)
```

![](assets/images/chapters/decontamination-authentication/cuperdec1.png)

> In the figure above, one curve represents one sample, and the red curves have a very high amount of contamination and very low amount of endogenous DNA. These samples might be considered to be dropped from the downstream analysis.

Reviewer: ```suggestion
In the figure above, one curve represents one sample, and the red curves have a very high amount of contamination and very low amount of endogenous DNA based on a given cut-off on the Y axis (cuperdec offers multiple methods for setting such cut-off). In the case of the blue lines, the majority of most abundant taxa in the sample are similar to those found in the expected 'source environment' (e.g. the oral cavity when dealing with dental calculus samples), whereas the the abundant taxa in the red samples from sources other than the expected environment (e.g. skin or soil taxa in calculus samples). These samples might be considered to be dropped from the downstream analysis.
```

## Source tracking

> For the case of ancient microbiome profiling, in addition to traditional inspection of the list of detected organisms and comparing it with the ones detected on blanks, it is useful to use pre-trained classifiers that can give a prediction on what environment the detected organisms most likely come from. The most popular and widely used tool is called [**SourceTracker**](https://www.nature.com/articles/nmeth.1650#citeas). SourceTracker is a Bayesian version of the Gaussian Mixture Model (GMM) clustering algorithm that is trained on a reference data called **Sources**, i.e. different classes such as Soil or Human Oral or Human Gut microbial communities etc., and then it can estimate proportion / contribution of each Source into test samples called **Sinks**.

Reviewer: ```suggestion
For the case of ancient microbiome profiling, in addition to traditional inspection of the list of detected organisms and comparing it with the ones detected on blanks, we can use tools that make a prediction on what environment the detected organisms most likely come from. 

The most popular and widely used tool is called [**SourceTracker**](https://www.nature.com/articles/nmeth.1650#citeas). SourceTracker is a Bayesian version of the Gaussian Mixture Model (GMM) clustering algorithm that is trained on a user-supplied reference data called **Sources**, i.e. different classes such as Soil or Human Oral or Human Gut microbial communities etc., and then it can estimate proportion / contribution of each of these sources the users actual samples called **Sinks**.
```

Reviewer: I just noted that throughout the decontamination section you're not really mentioning what type of input data are required (just imlpied...) e.g. for decontam requires taxonomic profiles and a metadta sheet, cuperdec also requries taxon tables etc., - a bit like how you've done in sourcetracker below!)

![](assets/images/chapters/decontamination-authentication/SourceTracker.png)

> Originally, SourceTracker was developed for 16S data, i.e. using only 16S ribosomal RNA genes, but it can be easily trained using also shotgun metagenomics data, which was demonstrated in its metagenomic extension called [mSourceTracker](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7100590/) and its faster and more scalable version [FEAST](https://www.nature.com/articles/s41592-019-0431-x). The input data for SourceTracker are metadata, i.e. each sample has to have "source" or "sink" annotation as well as environmental label (e.g. Oral, Gut, Soil etc.), and microbial abundances (OTU abundances) quantified in some way, for example through [QIIME](https://qiime2.org/) pipeline, MetaPhlan or Kraken. The SourceTracker R script can be downloaded from [https://github.com/danknights/sourcetracker](https://github.com/danknights/sourcetracker), it expects two input data frames: metadata with at least sample name, environment and source / sink labels, and abundance matrix. Note that source and sink metadata and abundances have to be merged together prior to using SourceTracker. Next, training SourceTracker on source samples and running predictions on sink samples can be done using following command lines:

Reviewer: ```suggestion
Originally, SourceTracker was developed for 16S data, i.e. using only 16S ribosomal RNA genes, but it can be easily trained using also shotgun metagenomics data, which was demonstrated in its metagenomic extension called [mSourceTracker](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7100590/) and its faster and more scalable version [FEAST](https://www.nature.com/articles/s41592-019-0431-x). The input data for SourceTracker are metadata, i.e. each sample has to have "source" or "sink" annotation as well as environmental label (e.g. Oral, Gut, Soil etc.), and microbial abundances (OTU abundances) quantified in some way, for example through [QIIME](https://qiime2.org/) pipeline, MetaPhlan or Kraken. The SourceTracker R script can be downloaded from [https://github.com/danknights/sourcetracker](https://github.com/danknights/sourcetracker).

 Sourcetracker expects two input data frames: metadata with at least sample name, environment and source / sink labels, and abundance matrix. Note that source and sink metadata and abundances have to be merged together prior to using SourceTracker. Next, training SourceTracker on source samples and running predictions on sink samples can be done using following command lines:
```

Note we are generally not really recommending people to download raw scripts from GitHub etc., but rather install via `conda` (As it's better with dependency management...) so I woudl recommend going through the whole thing and give conda commands instead as primary (you're welcome to also link to the raw code after).

results <- predict(st, otus[test.ix,], alpha1 = 0.001, alpha2 = 0.001)

# Prepare output for plotting
results$proportions <- results$proportions[order(-results$proportions[,"Oral"]),]
name <- rep(rownames(results$proportions), each = 4)
value <- as.numeric(t(results$proportions))
condition <- rep(c("Gut" , "Oral" , "Skin", "Unknown"), length(test.ix))
data <- data.frame(name, condition, value)

# Plot SourceTracker inference as a barplot
ggplot(data, aes(fill=condition, y=value, x=reorder(name,seq(1:length(name))))) + 
geom_bar(position = "fill", stat = "identity") + 
theme(axis.text.x = element_text(angle = 90, size = 5, hjust = 1, vjust = 0.5)) + 
xlab("Sample") + ylab("Fraction")
```

![](assets/images/chapters/decontamination-authentication/source_tracker.png)

In the figure above the SourceTracker was trained on Human Microbiome Project (HMP) data, and was capable of predicting the fractions of oral, gut, skin or other microbial composition on the query sink samples. In a similar way, environmental soil or marine microbes can be used as Sources. In this way, environmental percentage of contamination can be detected per sample.

> A drawback of SourceTracker, mSourceTracker and FEAST is that they require a microbial abundance table after a taxonomic classification with e.g. QIIME or Kraken has been performed. Such taxonomic classification can be biased since it is computed against a reference database with known taxonomic annotation. In contrast, a novel microbial source tracjing tool [decOM](https://www.biorxiv.org/content/10.1101/2023.01.26.525439v1.full) aims at moving away from database-dependent methods and using unsupervised approaches exploiting read-level sequence composition.

Reviewer: ```suggestion
A drawback of SourceTracker, mSourceTracker and FEAST is that they require a microbial abundance table after a taxonomic classification with e.g. QIIME or Kraken has been performed. Such taxonomic classification can be biased since it is computed against a reference database with known taxonomic annotation. In contrast, a novel microbial source tracking tool [decOM](https://www.biorxiv.org/content/10.1101/2023.01.26.525439v1.full) aims at moving away from database-dependent methods and using unsupervised approaches exploiting read-level sequence composition.
```

![](assets/images/chapters/decontamination-authentication/deCOM.png)

> *decOM* uses [kmtricks](https://academic.oup.com/bioinformaticsadvances/article/2/1/vbac029/6576015) to compute a matrix of k-mer counts on raw reads (fastq-files) from source samples, and then uses the source k-mer abundance matrix for looking up k-mer composition of sink samples. This allows *decOM* to calculate microbial contributions / fractions from the sources. For example, for estimating contributions from ancient Oral (aOral), modern Oral (mOral), Skin and Sediment / Soil environments one can use an already computed source matrix from here https://github.com/CamilaDuitama/decOM/ and provide it as a *-p_sources* parameter. Then *decOM* can be run using the following command line:

Reviewer: ```suggestion
*decOM* uses [kmtricks](https://academic.oup.com/bioinformaticsadvances/article/2/1/vbac029/6576015) to compute a matrix of k-mer counts on raw reads (FASTQ files) from source samples, and then uses the source k-mer abundance matrix for looking up k-mer composition of sink samples. This allows *decOM* to calculate microbial contributions / fractions from the sources. For example, for estimating contributions from ancient Oral (aOral), modern Oral (mOral), Skin and Sediment / Soil environments one can use an already computed source matrix from here [https://github.com/CamilaDuitama/decOM/](https://github.com/CamilaDuitama/decOM/) and provide it as a *-p_sources* parameter. Then *decOM* can be run using the following command line:
```

```R
df<-read.csv("decOM_output.csv", check.names=FALSE)

result <- subset(df, select = c("Sink", "Sediment/Soil", "Skin", 
"aOral", "mOral", "Unknown"))
rownames(result) <- result$Sink
result$Sink <- NULL
result <- result / rowSums(result)
result<-result[order(-result$aOral),]
head(result)

name <- rep(rownames(result), each = 5)
value <- as.numeric(t(result))
condition <- rep(c("Sediment/Soil","Skin","aOral","mOral","Unknown"), 
dim(result)[1])
data <- data.frame(name, condition, value)
head(data)

library("ggplot2")
library("viridis")
ggplot(data, aes(fill=condition, y=value, x=reorder(name,seq(1:length(name))))) + 
  geom_bar(position = "fill", stat = "identity") + 
  theme(axis.text.x = element_text(angle=90, size = 5, hjust = 1, vjust = 0.5)) + 
  xlab("Sample") + ylab("Fraction")
```

![](assets/images/chapters/decontamination-authentication/decOM.png)

> *decOM* has certain advantages compared to *SourceTracker* as its is a taxonomic classification / database free approach. However, it also appears to be very sensitive to the particular training / source data set. In the example above it can be seen that the microbial source tracking of sink samples is very much dominated by the Oral community, which was the training / source data set.

Reviewer: Missing a final summary output

It's also nice to include sections on:

- Questions to ask
- Useful literature

(see here as example: introduction-to-ngs-sequencing page)
